ingest/indexer/service/blockprocessor/block_updates_indexer_block_process_strategy_test.go

```diff
@@ -0,0 +1,155 @@
+package blockprocessor_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/osmosis-labs/osmosis/v26/app/apptesting"
+	commondomain "github.com/osmosis-labs/osmosis/v26/ingest/common/domain"
+	commonmocks "github.com/osmosis-labs/osmosis/v26/ingest/common/domain/mocks"
+	indexermocks "github.com/osmosis-labs/osmosis/v26/ingest/indexer/domain/mocks"
+	"github.com/osmosis-labs/osmosis/v26/ingest/indexer/service/blockprocessor"
+	sqsmocks "github.com/osmosis-labs/osmosis/v26/ingest/sqs/domain/mocks"
+)
+
+type BlockUpdateIndexerBlockProcessStrategyTestSuite struct {
+	apptesting.ConcentratedKeeperTestHelper
+}
+
+// TestBlockUpdateIndexerBlockProcessStrategyTestSuite verifies the block update indexer strategy for processing created pools.
+// The test suite initializes all supported pools (concentrated, cfmm, cosmwasm) via s.App.PrepareAllSupportedPools(), creating pool IDs 1-5.
+// Test cases inject pool creation data and validate expected behavior of the pair publisher.
+//
+// Scenarios tested:
+// - Happy path: single pool creation: should perform publishing
+// - Happy path: multiple pool creation: should perform publishing
+// - No pool creation: nothing is published
+// - Pool creation data without a match: still perform publishing but no creation data included
+func TestBlockUpdateIndexerBlockProcessStrategyTestSuite(t *testing.T) {
+	suite.Run(t, new(BlockUpdateIndexerBlockProcessStrategyTestSuite))
+}
+
+func (s *BlockUpdateIndexerBlockProcessStrategyTestSuite) TestPublishCreatedPools() {
+	tests := []struct {
+		name                             string
+		createdPoolIDs                   map[uint64]commondomain.PoolCreation
+		expectedPublishPoolPairsCalled   bool
+		expectedNumPoolsPublished        int
+		expectedNumPoolsWithCreationData int
+	}{
+		{
+			name: "happy path with one pool creation",
+			createdPoolIDs: map[uint64]commondomain.PoolCreation{
+				1: {
```

Comment:
Those seems quite repetitve, can we abstract creation of such objects and add helper methods for state mutation? For example: 

```
// returns new oject with defaults
func newObject() {
 return {
					PoolId:      1,
					BlockHeight: 12345,
					BlockTime:   time.Now(),
					TxnHash:     "txhash1",
				}
}
```

and then:

`newObject().WithPoolId(3)`

Reply:
Great suggestion! Done!